did-utils/src/methods/mod.rs

```diff
@@ -1,9 +1,78 @@
-pub mod common;
-pub mod errors;
-pub mod traits;
-
-pub mod did_key;
-pub mod did_peer;
-pub mod did_web;
+//! A collection of methods for DID resolution and related utilities.
+//!
+//! This module provides functionality for resolving Decentralized Identifiers (DIDs)
+//! using different DID methods including [`did:key`], [`did:web`], and [`did:peer`].
+//! 
+//! [`did:key`]: https://w3c-ccg.github.io/did-method-key/
+//! [`did:web`]: https://w3c-ccg.github.io/did-method-web/
+//! [`did:peer`]: https://identity.foundation/peer-did-method-spec/
+//! 
+//! # Examples
+//! 
+//! ### Basic did:key resolution example
+//! 
+//! ```
+//! # use did_utils::methods::{DIDResolver, DidKey};
+//! # use did_utils::methods::DIDResolutionOptions;
+//! 
+//! # async fn test_did_key() {
+//!     let did_key_resolver = DidKey::new();
+//!     let did = "did:key:z6MkhaXgBZDvotDkL5257faiztiGiC2QtKLGpbnnEGta2doK";
+//!     let output = did_key_resolver.resolve(did, &DIDResolutionOptions::default()).await;
+//! # }
+//! ```
+//! 
+//! ### An example demonstrating a basic usage of did:web
+//! 
+//! ```
+//! # use did_utils::methods::{DIDResolver, DidWeb};
+//! # use did_utils::methods::DIDResolutionOptions;
+//! 
+//! # async fn resolves_did_web_document() {
+//!     let port = 3000;
+//!     let host = "localhost";
+//! 
+//!     let formatted_string = format!("did:web:{}%3A{}", host.to_string(), port);
+//!     let did: &str = &formatted_string;
+//! 
+//!     let did_web_resolver = DidWeb::new();
+//!     let output = did_web_resolver.resolve(
+//!         did,
+//!         &DIDResolutionOptions::default()
+//!     ).await;
+//! # }
+//! 
+//! ### An example demonstrating a basic usage of did:peer
+//! 
+//! ```
+//! # use did_utils::methods::{DIDResolver, DidPeer};
+//! # use did_utils::methods::DIDResolutionOptions;
+//! 
+//! # async fn resolves_did_peer_document() {
+//!     let did = "did:peer:0z6MkhaXgBZDvotDkL5257faiztiGiC2QtKLGpbnnEGta2doK";
+//! 
+//!     let did_peer_resolver = DidPeer::new();
+//!     let output = did_peer_resolver.resolve(
+//!         did,
+//!         &DIDResolutionOptions::default()
+//!     ).await;
+//! # }
+//! ```
```

Comment:
I suggest to relocate these examples under their respective DID method modules.

Reply:
The respective modules are now private, and only their relevant public items are exported. This simplifies how we bring functionalities into scope. For example, to use `DidPeer` and all its related functionalities, we can now write:
```rust
use did-utils::methods::DidPeer;
```
which is more concise and consistent compared to the previous approach:
```rust
use did-utils::methods::did_peer::DidPeer;
```
This way, we follow the Rust standard library conventions, which abstract away implementation details and focus on the available functionalities and their usage. The `methods` module-level documentation now serves as the best place to demonstrate how to use the different methods. We can also move the examples to the respectives resolver functions and leave the user to figure out how to create and use the appropriate resolver.